tendme.lic

```diff
@@ -1,85 +1,86 @@
-# quiet
 =begin
   Documentation: https://elanthipedia.play.net/Lich_script_repository#tendme
 =end
 
-custom_require.call(%w[common common-healing])
+custom_require.call(%w[common common-healing events])
 
-def bind_open_wounds?
-  DRC.bput('heal', '^You')
-  pause 1
-  raw = reget(15)
-  bleeders = raw
-             .dup
-             .reverse
-             .take_while { |item| item !~ /^Bleeding/ }
-             .keep_if { |item| item =~ /^\s+(?:\w+|\w+\s\w+)\s{7}.+/ }
-             .delete_if { |item| item =~ /\bskin\b/ }
-  leeches = raw
-            .dup
-            .map { |x| x.scan(/leech on your [^\,\.]+,|leech on your [^\,\.]+\./).to_a }
-            .flatten
-            .compact
-  mites = raw
-          .dup
-          .map { |x| x.scan(/mite on your [^\,\.]+,|mite on your [^\,\.]+\./).to_a }
-          .flatten
-          .compact
-  lodged = raw
-           .dup
-           .map { |x| x.scan(/lodged \w* into your [^\,\.]+/).to_a }
-           .flatten
-           .compact
+class TendMe
 
-  bleeders = bleeders.keep_if { |item| !item.include?('tended') && !item.include?('inside') && !item.include?('Area') }
-
-  lodged.each do |message|
-    message =~ /into your ([^\,\.]+)/
-    DRCH.bind_wound(Regexp.last_match(1))
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'train', regex: /train/i, optional: true, description: 'unwrap and rebind wounds for optimum learning' },
+        { name: 'monitor', regex: /monitor/i, optional: true, description: 'run in background to keep bleeders tended' }
+      ]
+    ]
+    args = parse_args(arg_definitions)
+    waitrt? while bind_open_wounds?
+    monitor_health(args.train) if args.monitor
   end
 
-  bleeders.each do |message|
-    message =~ /^\s+(\w+|\w+\s\w+)\s{7}/
-    DRCH.bind_wound(Regexp.last_match(1))
+  def bind_open_wounds?
+    tended_wounds = false
+    health_data = DRCH.check_health
+    (
+      health_data['lodged'].values.flatten +
+      health_data['parasites'].values.flatten +
+      health_data['bleeders'].values.flatten
+    )
+    .reject { |wound| wound.internal? || wound.bleeding_rate =~ /tended|clotted/ } # untendable
+    .select do |wound|      
+      skilled_to_tend = DRCH.skilled_to_tend_wound?(wound.bleeding_rate)
+      unless wound.bleeding? && skilled_to_tend
+        DRC.message("You are not skilled enough to tend a #{wound.bleeding_rate.upcase} bleeder on a #{wound.body_part.upcase}, skipping")
+      end
+      # if not bleeding then it's a parasite or lodged item
+      !wound.bleeding? || skilled_to_tend
+    end
```

Comment:
Skip bleeders the character is not skilled to tend -- this avoids long RT and potentially making the wound worse